src/java.desktop/share/classes/javax/swing/border/EtchedBorder.java

```diff
@@ -136,17 +136,30 @@ public void paintBorder(Component c, Graphics g, int x, int y, int width, int he
 
         g.translate(x, y);
 
-        g.setColor(etchType == LOWERED? getShadowColor(c) : getHighlightColor(c));
-        g.drawRect(0, 0, w-2, h-2);
+        if (etchType == LOWERED) {
```

Comment:
So if you think we need to draw the darker colour - shadow - 2nd then please add a comment here.
Also I don't like the repeated blocks of code and I can't follow what they represent
could we have something like
drawFoo(Graphics g, Color, c, int w, int h) {
      g.setColor(c);
      g.drawLine(1, h-3, 1, 1);
      g.drawLine(1, 1, w-3, 1);

       g.drawLine(0, h-1, w-1, h-1);
      g.drawLine(w-1, h-1, w-1, 0);
}

drawBar(Graphics g, Color c, int w, int h) {
 g.setColor(c);
 g.drawRect(0, 0, w-2, h-2);
}

then it would become
if (etchType == LOWERED) {
  drawFoo(g, getHighlightColor(c), w, h);
  drawBar(g, getShadowColor(c), w, h);
} else {
  drawBar(g, getHighlightColor(c)w, h);
  drawFoo(g, getShadowColor(c), w, h); 

and of course we won't use "Foo" and "Bar" but something that actually represents what these do .. .which really isn't clear to me
Is "Bar" something like "Border" and "Foo" ... hmm I don't know.